callbacks.go

```diff
@@ -12,12 +12,24 @@ type AfterFindable interface {
 	AfterFind(*Connection) error
 }
 
-func (m *Model) afterFind(c *Connection) error {
-	if x, ok := m.Value.(AfterFindable); ok {
+// AfterEagerFindable callback will be called after a record, or records,
+// has been retrieved from the database and their associations have been
+// eagerly loaded.
+type AfterEagerFindable interface {
+	AfterEagerFind(*Connection) error
+}
+
+func (m *Model) afterFind(c *Connection, eager bool) error {
+	if x, ok := m.Value.(AfterFindable); ok && !eager {
 		if err := x.AfterFind(c); err != nil {
 			return err
 		}
 	}
+	if x, ok := m.Value.(AfterEagerFindable); ok && eager {
+		if err := x.AfterEagerFind(c); err != nil {
+			return err
+		}
+	}
```

Comment:
I could be tiny. Since those two blocks are exclusive to each other, I would prefer to rewrite the block for readability as below. What do you think? (same for blocks in line 49)

```go
	if eager {
		if x, ok := m.Value.(AfterFindable); ok {
			if err := x.AfterFind(c); err != nil {
				return err
			}
		}
	} else {
		if x, ok := m.Value.(AfterEagerFindable); ok {
			if err := x.AfterEagerFind(c); err != nil {
				return err
			}
		}
	}
```

Reply:
Basically just a habit of keeping nesting flat - would be fine either way :)